sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -28,14 +28,24 @@ abstract sealed class SortDirection {
   def sql: String
 }
 
+// default null order is firt for asc
 case object Ascending extends SortDirection {
   override def sql: String = "ASC"
 }
 
+case object AscendingNullLast extends SortDirection {
```

Comment:
Combining `SortDirection` with `NullOrdering` seems to make life more complicated then needed, why not keep null ordering a separate concept? And add that to `SortOrder`, for example:
```scala
case class SortOrder(child: Expression, direction: SortDirection, nullOrdering: NullOrdering) {
  def this(child: Expression, direction: SortDirection): SortOrder =
    this(child, direction, direction.DefaultNullOrdering)
  ...
}

abstract sealed class SortDirection {
  def sql: String
  def defaultNullOrdering: NullOrdering
}

case object Ascending extends SortDirection {
  override def sql: String = "ASC"
  override def defaultNullOrdering: NullOrdering = NullsFirst
}

case object Descending extends SortDirection {
  override def sql: String = "DESC"
  override def defaultNullOrdering: NullOrdering = NullsLast
}
```